packages/flutter_tools/test/integration.shard/test_driver.dart

```diff
@@ -173,15 +174,55 @@ abstract class FlutterTestDriver {
 
     _debugPrint('Sending SIGTERM to $_processPid..');
     ProcessSignal.SIGTERM.send(_processPid);
+    await _ensureDead(_processPid, 10);
     return _process.exitCode.timeout(quitTimeout, onTimeout: _killForcefully);
   }
 
-  Future<int> _killForcefully() {
+  Future<int> _killForcefully() async {
+    print('killing forcefully');
     _debugPrint('Sending SIGKILL to $_processPid..');
     ProcessSignal.SIGKILL.send(_processPid);
+    await _ensureDead(_processPid, 10);
     return _process.exitCode;
   }
 
+  // TODO(dnfield): This is racy. If dart-lang/sdk#40759 can be resolved, we
+  // should remove this.
+  Future<void> _ensureDead(int pid, int tries) {
```

Comment:
uber nit: shouldn't be called `ensureX` if it can't ensure it.

Reply:
Renamed to `tryWaitForPidDeath`, and moved it to test_utils.dart so it  can be reused in the daemon_mode_test for the same reason.

Reply:
SGTM